lib/iris/unit.py

```diff
@@ -1869,18 +1872,15 @@ def convert(self, value, other, ctype=FLOAT64):
                 ut_converter = _ut_get_converter(self.ut_unit, other.ut_unit)
                 if ut_converter:
                     if isinstance(value_copy, np.ndarray):
-                        # Can only handle array of np.float32 or np.float64 so
-                        # cast array of ints to array of floats of requested
+                        # Can only handle dtypes of np.float32 or np.float64 so
+                        # cast `value_copy` to floats of requested
                         # precision.
-                        if issubclass(value_copy.dtype.type, np.integer):
-                            value_copy = value_copy.astype(
-                                _ctypes2numpy[ctype])
+                        value_copy = value_copy.astype(_ctypes2numpy[ctype])
```

Comment:
Proposal: revert this change, byteswap if necessary, and improve the ctype keyword documentation to mention the fact that it is only for integer types.